### PR TITLE
[Refactor] refactor system prune flagging process

### PR DESCRIPTION
**Pull request by suyanhanx**
Signed-off-by: Han Xu <suyanhanx@gmail.com>

### CODE REVIEW

File: pkg/api/types/system_types.go [hunk 1]
```diff
@@ -38,3 +38,22 @@ type SystemEventsOptions struct {
 	// Format the output using the given Go template, e.g, '{{json .}}
 	Format string
 }
+
+// SystemPruneOptions specifies options for `nerdctl system prune`.
+type SystemPruneOptions struct {
+	Stdin  io.Reader
```
Comment (djdongjin) [inline pkg/api/types/system_types.go:44]: Is the `Stdin` used in `pkg` side? I think it's only used to confirm if `Force` or not.

File: pkg/cmd/system/prune.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		Stdout:               options.Stdout,
+	}); err != nil {
+		return err
+	}
+	if options.Volumes {
+		if err := volume.Prune(ctx, client, types.VolumePruneOptions{
+			GOptions: options.GOptions,
+			Force:    true,
+			Stdout:   options.Stdout,
+			Stdin:    options.Stdin,
```
Comment (djdongjin) [inline pkg/cmd/system/prune.go:56]: Not necessary in this PR, but we should move the `Force` confirmation logic in `volume.Prune` to `cmd` side to be consistent with other `Prune` and remove this `Stdin` option.

Comment (suyanhanx) [reply pkg/cmd/system/prune.go:56]: Agree. Will do this.

File: pkg/cmd/system/prune.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+			fmt.Fprintln(options.Stdout, item.ID)
+		}
+	}
+
+	// TODO: print total reclaimed space
+
+	return nil
+}
+
+func buildCachePrune(ctx context.Context, options types.SystemPruneOptions) ([]buildkitutil.UsageInfo, error) {
```
Comment (AkihiroSuda) [inline pkg/cmd/system/prune.go:84]: This should be in `pkg/cmd/builder`

Comment (suyanhanx) [reply pkg/cmd/system/prune.go]: Updated.

File: pkg/cmd/builder/prune.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	"io"
+	"os/exec"
+
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/buildkitutil"
+	"github.com/sirupsen/logrus"
+)
+
+// PruneBuildCache will prune all build cache.
+func PruneBuildCache(ctx context.Context, options types.SystemPruneOptions) ([]buildkitutil.UsageInfo, error) {
```
Comment (suyanhanx) [inline pkg/cmd/builder/prune.go:32]: Or just simply rename it to `Prune`?

File: pkg/api/types/system_types.go [hunk 5]
```diff
@@ -38,3 +38,21 @@ type SystemEventsOptions struct {
...
+// SystemPruneOptions specifies options for `nerdctl system prune`.
+type SystemPruneOptions struct {
+	Stdout io.Writer
+	Stderr io.Writer
+	// GOptions is the global options
+	GOptions GlobalCommandOptions
+	// All remove all unused images, not just dangling ones
+	All bool
+	// Force will not prompt for confirmation
+	Force bool
```
Comment (djdongjin) [inline pkg/api/types/system_types.go:51]: `Force` can be removed now right?

Comment (suyanhanx) [reply pkg/api/types/system_types.go]: Removed.

File: pkg/cmd/system/prune.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/containerd/containerd"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/cmd/builder"
+	"github.com/containerd/nerdctl/pkg/cmd/container"
+	"github.com/containerd/nerdctl/pkg/cmd/image"
+	"github.com/containerd/nerdctl/pkg/cmd/network"
+	"github.com/containerd/nerdctl/pkg/cmd/volume"
+)
+
+// Prune will remove all unused containers, networks, images (both dangling and unreferenced), and optionally, volumes.
```
Comment (djdongjin) [inline pkg/cmd/system/prune.go:32]: With https://github.com/containerd/nerdctl/pull/2036, `images (both dangling and unreferenced)` is no longer true (depends on the `All` value). You can update the comment in either one of the PRs.

**Review by djdongjin — APPROVED** [review]
LGTM, thanks

File: pkg/cmd/system/prune.go (hunk 6 above)
Comment (suyanhanx) [reply pkg/cmd/system/prune.go:32]: Yep.

Comment (suyanhanx) [reply pkg/cmd/system/prune.go]: Updated in this PR.

File: pkg/cmd/builder/prune.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	"io"
+	"os/exec"
+
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/buildkitutil"
+	"github.com/sirupsen/logrus"
+)
+
+// Prune will prune all build cache.
+func Prune(ctx context.Context, options types.SystemPruneOptions) ([]buildkitutil.UsageInfo, error) {
```
[the remark this answers is not in the document]
Comment (suyanhanx) [reply pkg/cmd/builder/prune.go]: 🤔 After this change list, will open another PR for refactoring `builder prune`. 
Updated.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**